editor.planx.uk/src/pages/FlowEditor/lib/analyticsProvider.tsx

```diff
@@ -410,6 +421,50 @@ export const AnalyticsProvider: React.FC<{ children: React.ReactNode }> = ({
     }
   }
 
+  async function updateLastVisibleNodeLogWithAllowListAnswers(nodeId: string) {
+    if (shouldTrackAnalytics && lastVisibleNodeAnalyticsLogId) {
```

Comment:
I think this is a great shout, I think theres a lot of nesting in the file which can make it a bit harder to parse. 

I've added more guard clauses and tried to see if I could update the conditionals to be slightly more readable. 

It's on this commit and it would be great to get your opinion on the changes: 7bb5ac814a8aae0640f010337eefc6bb160dc0c9

Reply:
This is nice and a good improvement - more readable for sure 👍 

What do you think about a change along these lines - 

```ts
function track(event: MyEnumOfEventTypes): void {
  skipUpdateIfNotTracking();

  switch (event) {
    case "helpClick":
      // Handle helpClick event
      break;
    case "nextStepsLinkClick":
      // Handle nextStepsLinkClick event
      break;
    // Add more cases as needed
    default:
      // Handle default case or throw an error
      break;
  }
}
```

The benefits I'd see of this would be - 
 - You can't miss the guard clause `skipUpdateIfNotTracking()` when adding a new event
 - It's a bit more self documenting - when we come back and add a new event type it should be a simpler pattern to follow (add event enum, add case in this controller function)
 
 Super happy for the response to be "not worth it", just wanted to run the idea by you. This could also be a small refactor on a follow up PR! 👍

Reply:
This is a really interesting suggestion! I'd noticed there was quite a lot of very similar patterns and something like this could potentially resolve that. Some of the events have specific requirements for further info but I'm sure this could be managed as well. 

As this piece of work has been on the burner for a long time I think I'd opt to as you say leave it for now and potentially follow up with a refactor PR in the future. I feel like I'm almost too close to the `analyticsProvider` file and a bit of time away from it could really help to refactor with fresh eyes